python/ml4ir/applications/ranking/tests/test_ml4ir_sanity.py

```diff
@@ -0,0 +1,266 @@
+    """
+    MRR = 0
+    queries = df.groupby('query_id')
+    for qid, group in queries:
+        X = [group['f' + str(i)] for i in range(featureCount)]
+        pred = model.decision_function(np.array(X).T)
+        mrr = calculate_MRR(pred, group['clicked'])
+        MRR += mrr
+    return MRR/len(queries)
+
+
+def check_accuracy(df, model, featureCount=2):
+    X = [df['f'+str(i)] for i in range(featureCount)]
+    Y = df['clicked']
+    acc = model.score(np.array(X).T, Y)
+    return acc
+
+
+def train_sklearn_model(df, model, featureCount=2):
+    """
+    Trains a sklearn model (Specifically, logistic regression and Perceptron)
+    """
+    X = [df['f' + str(i)] for i in range(featureCount)]
+    Y = df['clicked']
+    model.fit(np.array(X).T, Y)
+    for i in range(featureCount):
+        model.coef_[0][i] = 1.0
+    if model.fit_intercept:
+        model.intercept_ = 1.0
+    else:
+        model.intercept_ = 0
+    model.fit(np.array(X).T, Y)
+    return model
+
+
+def generate(args):
+    """
+    create an ml4ir consumable dataframe after generating dataset.
+    """
+    X, Y = generate_data(args)
+    df = create_ml4ir_dataframe(X, Y)
+    return df
+
+
+def wrap_linear_model_in_sklearn_model(df, W):
+    """
+    Wrap plain weights and bias in a dummy sklearn model
+    """
+    wrapper_model = Perceptron(fit_intercept=True, max_iter=1, verbose=0, validation_fraction=0.0001)
+    X = [df['f' + str(i)] for i in range(len(W)-1)]
+    Y = df['clicked']
+    wrapper_model.fit(np.array(X).T, Y)
+    for i in range(len(W)-1):
+        wrapper_model.coef_[0][i] = W[i]
+    wrapper_model.intercept_ = W[-1]
+    return wrapper_model
+
+
+def test_ml4ir_sanity_pipeline(args, p, log_dir):
+        df = generate(args)
+        df.to_csv(p / 'train' / 'data.csv')
+        df.to_csv(p / 'validation' / 'data.csv')
+        df.to_csv(p / 'test' / 'data.csv')
+
+        fconfig_name = "feature_config_sanity_tests_" + str(args.n_features) + "_features.yaml"
+        feature_config_file = pathlib.Path(__file__).parent / "data" / "configs" / fconfig_name
+        model_config_file = pathlib.Path(__file__).parent / "data" / "configs" / "model_config_sanity_tests.yaml"
+        train_ml4ir(p.as_posix(), feature_config_file.as_posix(), model_config_file.as_posix(), log_dir.as_posix())
+        logisticReg = LogisticRegression(fit_intercept=False, random_state=args.seed, max_iter=100,
+                                warm_start=True)
+        perceptron = Perceptron(fit_intercept=False, max_iter=100, random_state=args.seed, validation_fraction=0.0001,
+                       warm_start=True)
+        ml4ir_weights = pd.read_csv("models/test_command_line/coefficients.csv")["weight"].tolist()
+        ml4ir_weights.append(0.0)
+        ml4ir = wrap_linear_model_in_sklearn_model(df, ml4ir_weights)
+        train_sklearn_model(df, logisticReg, featureCount=args.n_features)
+        train_sklearn_model(df, perceptron, featureCount=args.n_features)
```

Comment:
@mohazahran We should definitely get rid of this logic here IMO. If I understand right, you are trying to load the trained model weights from ml4ir and then assign them + freeze a sklearn Perceptron. This is then used to compute MRR and accuracy. It is a lot of overkill I think. You can very well get all these metrics from `relevance_model.run_pipeline()` already.

CC: @jakemannix